### PR TITLE
Added Basic HTTP Informations

### PR DESCRIPTION
**Pull request by RichardLim00**
- Basic Concept of what is HTTP
- Describing the HTTP process using an example
- Added 2 Commonly Used Request Method description
- Added Brief Description and Types of Response Code

### CODE REVIEW

File: http/readme.md [hunk 1]
```diff
@@ -1,3 +1,42 @@
 # HyperText Transfer Protocol (HTTP)
 
+HTTP is a protocol **defines how client and server communicates**.
```
Comment (DanielOrmeno) [inline http/readme.md:3]: HTTP is a protocol `that` **defines how client and server communicates**. 

Maybe you can add a link to the Client-Server topic of this module! :D

File: http/readme.md [hunk 2]
```diff
@@ -1,3 +1,42 @@
...
+HTTP is a protocol **defines how client and server communicates**. 
+
+Everytime we search up something with our browser, let say Wikipedia, the browser will then communicate with Wikipedia's server to get the resources you had asked for.
+
+Let's see how the communication process works.
+  1. The **Browser send the request** for resources to the server. (in the form of URL)
+  2. The **Server process the request**, then **respond** accordingly to it.
+  
+### Examples
+**Request** and **Response** are the keywords in HTTP. To make it easier, imagine the communication between server and client as conversation between a Customer and a Bartender.
```
Comment (DanielOrmeno) [inline http/readme.md:12]: Nice analogy! could we change it from bartender to maybe waiter? Keeping it friendly for potential young students.

**Review by DanielOrmeno — CHANGES_REQUESTED** [review]
Thank you for your contribution! This is very cool. I've left a couple of comments requesting some minor changes.

Could you please also run your text through a grammar checker just to make sure that the language is accurate? I found a few small places for improvement. I can suggest grammarly or a similar tool. 

Thanks again for your contribution! looking forward for the next revision

**Comment by DanielOrmeno** [comment]
I'll merge this PR so that others can improve the content. Please submit a new PR with the requested changes.

File: http/readme.md (hunk 2 above)
Comment (RichardLim00) [reply http/readme.md:12]: Sure! and Thank You!

**Pull request merged**